### PR TITLE
Document FMA settings

### PR DESCRIPTION
**Pull request by yxsamliu**

### CODE REVIEW

File: docs/markdown/hip_programming_guide.md [hunk 1]
```diff
@@ -115,4 +115,13 @@ allocated.
...
 
+## FMA
+
+By default HIP-Clang assumes -ffp-contract=fast and HCC assumes -ffp-contract=off.
+For x86_64, FMA is off by default since the generic x86_64 target does not
+support FMA by default. To turn on FMA on x86_64, either use -mfma or -march=native
+on CPU's supporting FMA.
+
+When CPU and GPU has different FMA settigns, the results may not equal. Tolerance
+should be used for floating point comparsion.
```
Comment (emankov) [inline docs/markdown/hip_programming_guide.md]: LGTM, otherwise add a new line.

Comment (yxsamliu) [reply docs/markdown/hip_programming_guide.md]: fixed

File: docs/markdown/hip_programming_guide.md [hunk 2]
```diff
@@ -115,4 +115,14 @@ allocated.
...
 In HCC and HIP-Clang, long double type is 80-bit extended precision format for x86_64, which is not supported by AMDGPU. HCC and HIP-Clang treat long double type as IEEE double type for AMDGPU. Using long double type in HIP source code will not cause issue as long as data of long double type is not transferred between host and device. However, long double type should not be used as kernel argument type.
 
+## FMA
+
+By default HIP-Clang assumes -ffp-contract=fast and HCC assumes -ffp-contract=off.
+For x86_64, FMA is off by default since the generic x86_64 target does not
+support FMA by default. To turn on FMA on x86_64, either use -mfma or -march=native
+on CPU's supporting FMA.
+
+When CPU and GPU has different FMA settigns, the results may not equal. Tolerance
```
Comment (dmcdougall) [inline docs/markdown/hip_programming_guide.md]: > `settigns`

That's a typo.

File: docs/markdown/hip_programming_guide.md (hunk 1 above)
Comment (b-sumner) [reply docs/markdown/hip_programming_guide.md]: For the second paragraph, I'd suggest: "When contractions are enabled and the CPU has not enabled FMA instructions, the GPU can produce different numerical results than the CPU for expressions that can be contracted."

File: docs/markdown/hip_programming_guide.md [hunk 3]
```diff
@@ -115,4 +115,14 @@ allocated.
 
 In HCC and HIP-Clang, long double type is 80-bit extended precision format for x86_64, which is not supported by AMDGPU. HCC and HIP-Clang treat long double type as IEEE double type for AMDGPU. Using long double type in HIP source code will not cause issue as long as data of long double type is not transferred between host and device. However, long double type should not be used as kernel argument type.
 
+## FMA
```
Comment (b-sumner) [inline docs/markdown/hip_programming_guide.md]: I suggest naming this section "FMA and contractions"

File: docs/markdown/hip_programming_guide.md (hunk 2 above)
Comment (yxsamliu) [reply docs/markdown/hip_programming_guide.md]: typo removed. thanks.

**Comment by yxsamliu** [comment]
updated by Brian's suggestion. thanks.

**Pull request merged**